tests/test_multipart_operations.py

```diff
@@ -424,3 +444,97 @@ def test_copy_obj_to_another_bucket_with_different_fs_options(
         s3_client.download_bucket_contents(second_bucket_name, results_dir)
         assert check_data_integrity(origin_dir, results_dir)
         log.info("Both uploaded and copied data are identical")
+
+    @tier1
+    def test_multipart_upload_on_versioned_bucket(
+        self,
+        c_scope_s3client,
+        tmp_directories_factory,
+    ):
+        """
+        Test basic s3 operations using a noobaa bucket:
+        1. Create bucket and enable versioning on it
+        2. Upload 5 copies of data with same key with different chunk size
+        3. List all available versions of uploaded data
+        4. Download each copy and compare it with original data
+        5. Create new non-versioned bucket and copy object in it
+        6. Verify copied data is identical with uploaded data
+        """
+        # 1. Create a bucket using S3 and enable versioning on it
+        bucket_name = c_scope_s3client.create_bucket()
+        c_scope_s3client.put_bucket_versioning(bucket_name, status="Enabled")
+        log.info(f"Enabled bucket versioning on bucket {bucket_name}")
+        log.info("Uploading multipart object")
+        origin_dir, results_dir = tmp_directories_factory(
+            dirs_to_create=["origin", "result"]
+        )
+        # 2. Write multipart objects to the bucket
+        object_name = generate_random_files(
+            origin_dir,
+            1,
+            min_size="20M",
+            max_size="30M",
+        )
```

Comment:
Simplify the rest by indexing ther result here instead of using `object_name[0]` later. The main issue for readability/maintenance is that `object_name` appears as a list.
```suggestion
        object_name = generate_random_files(
            origin_dir,
            1,
            min_size="20M",
            max_size="30M",
        )[0]
```

Reply:
Fixed